modules/lang-painless/src/test/java/org/elasticsearch/painless/ScriptTestCase.java

```diff
@@ -114,10 +116,30 @@ public void assertBytecodeHasPattern(String script, String pattern) {
 
     /** Checks a specific exception class is thrown (boxed inside ScriptException) and returns it. */
     public static <T extends Throwable> T expectScriptThrows(Class<T> expectedType, ThrowingRunnable runnable) {
+        return expectScriptThrows(expectedType, true, runnable);
+    }
+
+    /** Checks a specific exception class is thrown (boxed inside ScriptException) and returns it. */
+    public static <T extends Throwable> T expectScriptThrows(Class<T> expectedType, boolean shouldHaveScriptStack,
+            ThrowingRunnable runnable) {
         try {
             runnable.run();
         } catch (Throwable e) {
             if (e instanceof ScriptException) {
+                boolean hasEmptyScriptStack = ((ScriptException) e).getScriptStack().isEmpty();
```

Comment:
I was originally hunting down a case where we weren't producing the stack. I never found it but I figure something like this is nice to have around anyway.